Service/src/main/java/Reader/ReadFile.java

```diff
@@ -0,0 +1,54 @@
+package Reader;
+
+import Validator.Validator;
+import org.example.SetClassId;
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.math.BigDecimal;
+import java.time.LocalTime;
+import java.util.ArrayList;
+import java.util.List;
+import TIcket.ticketTypes;
+import TIcket.Ticket;
+import java.time.LocalDate;
+
+public class ReadFile implements SetClassId {
+    private static int classId;
+
+    public static List<Ticket> readTicketsFromFile(String filePath) {
+        List<Ticket> tickets = new ArrayList<>();
+
+        try (BufferedReader br = new BufferedReader(new java.io.FileReader(filePath))) {
+            String line;
+            int i = 1;
+            while ((line = br.readLine()) != null) {
+                String[] parts = line.split(",");
+                LocalDate creationDate = LocalDate.parse(parts[0]);
+                LocalTime creationTime = LocalTime.parse(parts[1]);
+                String typeOfTicket = parts[2].toUpperCase();
+                BigDecimal cost = new BigDecimal(parts[3]);
+                Validator validator = new Validator();
+                if(validator.isValidTicket(creationDate, typeOfTicket, cost)){
+                    ticketTypes ticketType = ticketTypes.valueOf(parts[2].toUpperCase());
+                    tickets.add(new Ticket(creationDate, creationTime, ticketType, cost));
+                    System.out.println("Ticket " + i + " was successfully read");
+                }else{
+                    System.out.println("Error. Invalid ticket data. Ticket " + i + " cannot be read");
+                }
+                i++;
+            }
+        } catch (IOException e) {
+            e.printStackTrace();
+        }
+        return tickets;
+    }
+    @Override
+    public void setId(int id){
```

Comment:
does reading of file connecting with setting id? Why do it implemented setClassId?

Reply:
No it isn't, but in our previous task we have got a paragraph "Any class in your application should be able to have a numeric "ID" field and be able to get and set it..." I thought the previous requirements were valid for all homework. It is implemented setClassId because 
setting and getting an ID is a common action that all classes must perform.

Reply:
Each class should nave an opportunity to have ... No needs to actual implementing/extending it :)